Pull request for reviewing and merging the initial fixes.

rahulait opened the pull request.

rahulait committed:
Initial commit of ssl changes.

Changes include:-
1. Addition of different flags to enable ssl for different services.
2. Fixing the sequence of execution of various files/packages/services.
3. Added support for self signed certificate, installing it automatically.
4. Fixed puppet scripts to allow automation of ssl.

Will fix the redundant code committed in this patch in future fixes.

rahulait committed:
Modification of puppet files to support ssl.

Added various missing flags and automated the procedure.

rahulait commented on glance/manifests/api.pp:
```diff
@@ -239,6 +239,21 @@
+    class {'moc_openstack::ssl::add_glance_cert':
+      require => File['/etc/glance'],
+    }
+    class {'moc_openstack::ssl::temp_glance_fix':
```
It fails for compute since /etc/glance is not present on compute. Need to find a better fix for this.

rahulait replied:
Added the fix. Currently, I am not updating the glanceclient version from 0.17 to 0.18 or more since its requiring many other packages to be updated (required me to install gcc, python-devel and libffi which I don't want). Hence, I am keeping the current patch, we will remove it once we have all the updated packages.

rahulait commented on moc_openstack/manifests/ssl/add_cinder_cert.pp:
```diff
@@ -0,0 +1,23 @@
+class moc_openstack::ssl::add_cinder_cert (
+  $cinder_crt      = "/etc/pki/tls/certs/cinder.crt",
+  $cinder_key      = "/etc/pki/tls/private/cinder.key",
+  $cinder_crt_path = "puppet:///modules/quickstack/certs/host.crt",
```
Should I keep the path for files as modules/quickstack/files or should I change it to modules/moc_openstack/files ? This comment applies to all the classes using moc_openstack::ssl.

lokI8 replied:
You should move it to moc_openstack. We don't want to put any of our moc specific files in the quickstack module.

rahulait commented on moc_openstack/manifests/ssl/additional_params.pp:
```diff
@@ -0,0 +1,66 @@
+# Apply ssl related changes to controller node
```
This file is not used at this moment. All the changes from this file are moved to individual service modules. Keeping those changes here was causing timing issues (like file not present and we trying to modify it).

rahulait commented on moc_openstack/manifests/ssl/temp_glance_fix.pp:
```diff
@@ -0,0 +1,14 @@
+  $file   = "/usr/lib/python2.7/site-packages/glanceclient/common/https.py",
+  $source = "puppet:///modules/quickstack/https.py",
+) {
+
```
Should we do this or should we pull the updated glanceclient rpm and put it on the localserver?

rahulait replied:
Updated glanceclient caused different other issues. I used "pip install python-glanceclient==0.18 --upgrade" or some other version > 0.17, but it was causing various other dependency issues. Hence, left it as it is.

rahulait commented on quickstack/manifests/admin_client.pp:
```diff
@@ -25,7 +25,7 @@
 export OS_PASSWORD=$admin_password
-export OS_AUTH_URL=http://$controller_admin_host:35357/v2.0/
+export OS_AUTH_URL=https://$controller_admin_host:35357/v2.0/
 export PS1='[\\u@\\h \\W(openstack_admin)]\\$ '
```
Add a check for updating to https only when we have configured ssl.

rahulait replied:
Added the fix.

rahulait commented on quickstack/manifests/compute_common.pp:
```diff
@@ -76,6 +89,25 @@
+    if str2bool_i("$amqp_ssl") {
+      $qpid_protocol = 'ssl'
+      $real_amqp_port = $amqp_ssl_port
+    }
```
Add else section to use 'tcp'

rahulait commented on quickstack/manifests/compute_common.pp:
```diff
@@ -76,6 +89,25 @@
+
+    if str2bool_i("$mysql_ssl") {
+      $nova_sql_connection = "mysql://nova:${nova_db_password}@${mysql_host}/nova?ssl_ca=${mysql_ca}"
+    }
```
Fix and add else section if we are not using ssl for mysql. Edit the other files as well wherever such a check has been made.

rahulait commented on quickstack/manifests/cinder.pp:
```diff
@@ -27,6 +27,14 @@
+  $auth_uri        = 'http://localhost:5000/v2.0',
+  $identity_uri    = 'http://localhost:35357/v2.0',
+  $auth_protocol   = 'http',
+  $cert_file       = '/etc/pki/tls/certs/cinder.crt',
```
check if its possible to remove auth_protocol and determine it based on "use_ssl" flag.

rahulait replied:
Done.

rahulait commented on quickstack/manifests/compute_common.pp:
```diff
@@ -62,9 +62,22 @@
+  $glance_priv_url              = $quickstack::params::glance_priv_url,
+  $use_ssl                      = $quickstack::params::use_ssl_endpoints,
+  $auth_protocol                = $quickstack::params::auth_protocol,
+  $enabled_ssl_apis             = ['osapi_compute'],
```
determine auth_protocol based on use_ssl flag

rahulait replied:
fixed.

rahulait commented on quickstack/manifests/neutron/compute.pp:
```diff
@@ -74,12 +78,24 @@
+      $qpid_protocol = 'ssl'
+      $real_amqp_port = $amqp_ssl_port
+    }
+    if str2bool_i("$mysql_ssl") {
```
handle false scenario

rahulait replied:
Added.

rahulait committed:
Fixed compute_common.pp to include temp_glance_fix.

1. Added temporary glance fix for the https bug.
2. Handled the else case failures.

rahulait commented on quickstack/manifests/controller_common.pp:
```diff
@@ -129,61 +135,117 @@
+  $ceph_nodes			               = $quickstack::params::ceph_nodes,
+  $ceph_enpoints		             = $quickstack::params::ceph_endpoints,
+  $ceph_user			               = $quickstack::params::ceph_user,
   $ceph_vlan                     = $quickstack::params::ceph_vlan,
```
Removed the tabs here....

rahulait commented on quickstack/manifests/neutron/controller.pp:
```diff
@@ -171,12 +174,30 @@
+  $neutron_pub_url               = $quickstack::params::neutron_pub_url,
+  $keystone_admin_url            = $quickstack::params::keystone_admin_url,
+  $auth_protocol                 = $quickstack::params::auth_protocol,
+  $ovs_l2_population             = 'true',
```
Remove auth_protocol.

rahulait replied:
Done.

rahulait commented on quickstack/manifests/neutron/networker.pp:
```diff
@@ -35,9 +38,13 @@
+    if str2bool_i("$mysql_ssl") {
+      $sql_connection = "mysql://neutron:${neutron_db_password}@${mysql_host}/neutron?ssl_ca=${mysql_ca}"
+    }
   } else {
```
Handle false cases.

rahulait replied:
Done.

rahulait commented on quickstack/manifests/nova.pp:
```diff
@@ -147,13 +147,21 @@
   $cpu_allocation_ratio         = '16',
   $verbose                      = 'false',
+  $use_ssl_endpoints            = $quickstack::params::use_ssl_endpoints,
+  $neutron_url                  = 'http://127.0.0.1:9696',
```
Changed use_ssl_endpoints to use_ssl to make it consistent with other files.

rahulait commented on quickstack/manifests/storage_backend/cinder.pp:
```diff
@@ -33,14 +33,21 @@
+    if str2bool_i("$mysql_ssl") {
+      $sql_connection = "mysql://cinder:${cinder_db_password}@${mysql_host}/cinder?ssl_ca=${mysql_ca}"
+    }
   } else {
```
handled false scenarios.

rahulait committed:
Fixed according to comments provided.

Commit:
Deleted the empty directory[temporary deletion]

Commit:
Added the openstack module which was missing.

Commit:
Moved the certificates location.

Changed it from quickstack/files to moc_openstack/files.

rahulait commented on cinder/manifests/init.pp:
```diff
@@ -261,6 +261,9 @@
+  $auth_uri                    = 'http://localhost:5000/',
+  $identity_uri                = 'http://localhost:35357/',
+  $nova_pub_url                = 'http://localhost:8776/',
   # DEPRECATED PARAMETERS
```
move them to api.pp and overwrite the default values by specifying in hiera.yaml file.

rahulait commented on cinder/manifests/init.pp:
```diff
@@ -472,8 +482,19 @@
+      'DEFAULT/nova_endpoint_template': value => $nova_pub_url;
+      'keystone_authtoken/auth_protocol': value => 'https';
+      'keystone_authtoken/auth_uri': value => $keystone_auth_uri;
+      'keystone_authtoken/identity_uri': value => $keystone_identity_uri;
```
can we remove auth_uri and identity_uri and check what happens? Ideally, they should not be required.

rahulait committed:
Fixed timing issue for l2_population configuration.

rahulait committed:
Revert "Fixed timing issue for l2_population configuration."

This reverts commit d0e4856d273894b50a1213a0ea4a503366abbf3e.

rahulait committed:
Fixed timing issue for l2_population in neutron.

rahulait committed:
Merge branch 'master' of https://github.com/rahulait/kilo-puppet

rahulait committed:
Fix neutron-configuration while pupetizing.

Added fix to allow self-signed certificates. Earlier, it
used to fail if the certificate is self signed, now it works
properly as its ignored. This is a temporary fix, we need a proper
fix where we pass the ca_file.

rahulait committed:
Changed authtoken to uuid instead of pki.

rahulait committed:
Configuring parameters from hiera.yaml.

Updated the init file not to have the configuration changes for
auth_uri and identity_uri. Instead, we use the parameters of api.pp
and overwrite the default values using hiera file.

rahulait committed:
Fixed the failures. Changes include:-

1. Fix for cinder timing issues.
2. Fix for neutron ovs configuration failure during first time.
3. Fix for temporary glance fix ordering.
4. Fix for configuring cinder parameters directly from hiera file.

rahulait committed:
Fixed few variable issues.

rahulait committed:
Fixed issues in compute files.

lokI8 commented:
+1. It's functional, but there are a couple things that could be made nicer. Going to take out bugs on them.

ravisantoshgudimetla commented:
+1. As discussed if we could abstract things out like having a single class with parameters that would be awesome.

gsilvis commented on nova/manifests/compute.pp:
```diff
@@ -131,6 +131,9 @@
   $pci_passthrough                    = undef,
+  $auth_protocol                      = 'http',
+  $auth_uri                           = 'http://127.0.0.1:5000/v2.0',
+  $identity_uri                       = 'http://127.0.0.1:35357',
```
I'm confused---these are Keystone ports, so I'm assuming that this is for nova-compute's API calls to Keystone.  Why are we using HTTP here, and why is the address localhost?

rahulait replied:
These are the default values. These get overwritten with the values we are passing to this class. For default values, the puppet modules use http and localhost.

gsilvis replied:
All right, then, looks good to me.

gsilvis commented:
+1
